lib/student.rb

```diff
@@ -6,6 +6,7 @@ class Student < Person
   def initialize(age, classroom, name = 'Unknown', parent_permission: true)
     super(age, name, parent_permission)
     @classroom = classroom
+    classroom&.add_student(self)
```

Comment:
According to the project requirements, please, create `a method` that serves as the 'belongs-to' side (a student belongs to a classroom). 

![image](https://github.com/ITurres/OOP_school_library/assets/78251927/588fa3b4-f0ef-4471-a209-40c0a0832f77)

So any time we add a student to a classroom, the classroom of the student is set as well, and vice versa (Both ways relationships). 💪🛠

Link to the [article with examples](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships)

Reply:
Hi @abdulhamiid, Thank you for your diligent review of this PR! Your insights are much appreciated. Let's dive into the changes suggested! 🎯 🙏🏻

The 
```rb
 classroom&.add_student(self)
```
makes use of the **`&.`** operator for safe navigation, allowing to call a method on an object if the object is not **_nil_**. In this case, it's calling the `'add_student'` method on the classroom object if **`-ClassRoom-`** is not **_nil_**.
This already handles the relationship between **`-Student-`** and **`-ClassRoom-`** by using the `'add_student'` method within the **`-ClassRoom-`** class.
The code establishes a `belongs-to` relationship. So when a new Student is created and added to a **`-ClassRoom-`**, the student's classroom attribute is set to the corresponding classroom, and the student is added to the classroom's list of students. This indicates that the **`-Student-`** class belongs to a **`-ClassRoom-`** class.

I hope this is of help! 🤠